Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -1337,6 +1319,52 @@ theorem bddAbove_range_comp {ι : Type u} {f : ι → Ordinal.{v}} (hf : BddAbov
   rw [bddAbove_iff_small] at hf
   exact small_lift _
 
+  -- TODO: generalize universes, make sSup version.
+theorem IsNormal.map_iSup {f : Ordinal.{u} → Ordinal.{max u v}} (H : IsNormal f) {ι : Type w}
+    (g : ι → Ordinal.{u}) [Nonempty ι] [Small.{u} ι] : f (⨆ i, g i) = ⨆ i, f (g i) :=
+  eq_of_forall_ge_iff fun a ↦ by
+    change f (⨆ i, g i) ≤ a ↔ ⨆ i, (f ∘ g) i ≤ a
+    rw [ciSup_le_iff (Ordinal.bddAbove_range_comp (bddAbove_of_small _) f)]
+    simp_all only [Function.comp]
+    rw [H.le_set' Set.univ Set.univ_nonempty g] <;>
```

Comment:
Something else that would be nice to do (in this PR or otherwise) is deprecate `IsNormal.le_set`. I think all it's doing is state `f (Sup s) = Sup (f '' s)` in a more roundabout way.

Reply:
We already have [`small_of_injective`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Logic/Small/Basic.html#small_of_injective).

Reply:
`small_of_injective` is not enough, `Small.{v + 1} Ordinal.{u}` doesn't imply `Small.{v} α` for `α` small in `u` ([zulip](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Smallness.20inequality)). So I added a few `Small` stuff.